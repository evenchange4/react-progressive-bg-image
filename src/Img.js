import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledImg = styled.div`
  height: 100%;
  background-repeat: no-repeat;
  transition: ${props => (props.isCached ? 'none' : props.transition)};

  opacity: ${props => (props.isLoaded ? 1 : props.opacity)};
  filter: ${props => (props.isLoaded ? 'none' : `blur(${props.blur}px)`)};
  /* this is needed so Safari keeps sharp edges */
  transform: ${props => (props.isLoaded ? 'none' : `scale(${props.scale})`)};
`;

const Img = ({ image, style, ...otherProps }) =>
  <StyledImg
    style={{
      ...style,
      backgroundImage: `url("${image}")`,
    }}
    {...otherProps}
  />;

Img.displayName = 'Img';
Img.propTypes = {
  // Internal
  image: PropTypes.string.isRequired,
  isLoaded: PropTypes.bool.isRequired,
  isCached: PropTypes.bool.isRequired,

  // props
  opacity: PropTypes.number.isRequired,
  blur: PropTypes.number.isRequired,
  scale: PropTypes.number.isRequired,
  className: PropTypes.string,
  transition: PropTypes.string,
  style: PropTypes.object,
};

export default Img;
