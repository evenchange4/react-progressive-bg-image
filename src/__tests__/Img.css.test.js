import 'jest-styled-components';
import React from 'react';
import { mount } from 'enzyme';
import Img from '../Img';

it('should render <Img> with isLoaded prop', () => {
  const component = (
    <Img
      image="image.jpg"
      isLoaded
      blur={2}
      opacity={0.9}
      scale={1}
      isCached
      component="div"
    />
  );

  expect(mount(component)).toMatchSnapshot();
});

it('should render <Img> without isLoaded prop', () => {
  const component = (
    <Img
      image="image.jpg"
      isLoaded={false}
      blur={2}
      opacity={0.9}
      scale={1}
      isCached={false}
      component="img"
    />
  );

  expect(mount(component)).toMatchSnapshot();
});
