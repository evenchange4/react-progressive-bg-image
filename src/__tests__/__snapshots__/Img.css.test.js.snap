// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <Img> with isLoaded prop 1`] = `
<styled.div
  blur={2}
  isLoaded={true}
  opacity={0.9}
  scale={1}
  style={
    Object {
      "backgroundImage": "url(\\"image.jpg\\")",
    }
  }
/>
`;

exports[`should render <Img> without isLoaded prop 1`] = `
<styled.div
  blur={2}
  isLoaded={false}
  opacity={0.9}
  scale={1}
  style={
    Object {
      "backgroundImage": "url(\\"image.jpg\\")",
    }
  }
/>
`;
